pymc3/distributions/continuous.py

```diff
@@ -2780,46 +2789,28 @@ class HalfStudentT(PositiveContinuous):
         with pm.Model():
             x = pm.HalfStudentT('x', lam=4, nu=10)
     """
+    rv_op = halfstudentt
+
+    @classmethod
+    def dist(cls, nu=1, sigma=None, lam=None, sd=None, *args, **kwargs):
 
-    def __init__(self, nu=1, sigma=None, lam=None, sd=None, *args, **kwargs):
-        super().__init__(*args, **kwargs)
         if sd is not None:
             sigma = sd
 
-        self.mode = at.as_tensor_variable(0)
         lam, sigma = get_tau_sigma(lam, sigma)
-        self.median = at.as_tensor_variable(sigma)
-        self.sigma = self.sd = at.as_tensor_variable(sigma)
-        self.lam = at.as_tensor_variable(lam)
-        self.nu = nu = at.as_tensor_variable(floatX(nu))
+        mode = at.as_tensor_variable(0)
```

Comment:
everything that is not going into `dist` can be commented for the time being. This is related to sampler starting point logic that has not been refactored.